File: appkey-service/src/main/java/org/appkey/config/ApplicationConfig.java
package org.appkey.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.cassandra.config.AbstractCassandraConfiguration;
import org.springframework.data.cassandra.repository.config.EnableCassandraRepositories;

@Configuration
@EnableCassandraRepositories
class ApplicationConfig extends AbstractCassandraConfiguration {

    //@Value("${spring.data.cassandra.port}")
    //public int port;

    @Value("${spring.data.cassandra.contact-points}")
    public String contactPoints;

    @Value("${spring.data.cassandra.local-datacenter}")
    public String datacenter;

    @Value("${spring.data.cassandra.keyspace-name}")
    public String keyspace;

    @Override
    public int getPort() {
        return 9042;
    }

    @Override
    public String getContactPoints() {
        return contactPoints;
    }


    @Override
    public String getLocalDataCenter() {
        return datacenter;
    }

    @Override
    protected String getKeyspaceName() {
        return keyspace;
    }

    public String[] getEntityBasePackages() {
        return new String[] { "org.appkey" };
    }

}

File: appkey-service/src/main/kotlin/org/appkey/application/config/ApplicationConfig.kt
package org.appkey.application.config

import org.springframework.data.cassandra.repository.config.EnableCassandraRepositories
import org.springframework.data.cassandra.config.AbstractCassandraConfiguration
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration

@Configuration
@EnableCassandraRepositories
class ApplicationConfig : AbstractCassandraConfiguration() {
    //@Value("${spring.data.cassandra.port}")
    //public int port;
    @Value("\${spring.data.cassandra.contact-points}")
    private lateinit var  contactPoints: String

    @Value("\${spring.data.cassandra.local-datacenter}")
    private lateinit var  datacenter: String

    @Value("\${spring.data.cassandra.keyspace-name}")
    private lateinit var  keyspace: String

    public override fun getPort(): Int {
        return 9042
    }

    public override fun getContactPoints(): String {
        return contactPoints
    }

    public override fun getLocalDataCenter(): String {
        return datacenter
    }

    override fun getKeyspaceName(): String {
        return keyspace
    }

    override fun getEntityBasePackages(): Array<String> {
        return arrayOf("org.appkey")
    }
}